encounter.md

```diff
@@ -0,0 +1,258 @@
+
+* Any healthcare worker that a patient can have an encounter with is a provider.
+
+## Available operations. 
+
+1. [List encounters](#list-encounters)
+2. [Create an encounters](#create-a-encounters)
+3. [Update an encounters](#update-a-encounters)
+4. [Delete an encounters](#delete-a-encounters)
+5. [List encounter provider sub resource](#list-encounter-provider-sub-resources)
+6. [Create encounter provider  sub resource with properties](#create-encounter-provider-sub-resource-with-properties)
+7. [Update encounter provider  sub resource](#update-encounter-provider-sub-resource)
+6. [Delete encounter provider  sub resource](#delete-encounter-provider-sub-resource)
+
+
+### List encounters
+
+* #### List all non-retired encounters.
+    
+    Quickly filter encounters with given query parameters.Returns a `404 Not Found` status if encounter not exists. If user not logged 
+    in to perform this action,a `401 Unauthorized` status returned.
+    
+    ##### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *q* | `Search Query` | Get encounter by Encounter Id, Patient Identifier or name
+    *patient* | `Patient UUID` | Get encounter by target_patient_uuid
+    *encounterType* | `Encounter_Type UUID` | Encounter type uuid. Must be used with patient
+    *order* | `Order UUID` | Filter by target_order_uuid. Must be used with patient
+    *obsConcept* | `Obs UUID` | Filter by target_obs_uuid. Must be used with patient
+    *obsValues* | `String` | Filter vy value of the obs object. Must be used with patient & obsConcept.
+    *fromdate* | `Date (ISO8601 Long)` | Start date of the encounter type. Must be used with patient
+    *todate* | `Date (ISO8601 Long)` | End date of the encounter type. Must be used with patient
+    
+
+    ```console
+    GET /encounter?
+      q=John
+      &fromdate=2016-10-08T04:09:23.000Z
+     ```
+    
+* #### List encounter by UUID.
+
+    Retrieve an encounter by its UUID. Returns a `404 Not Found` status if encounter not exists. If user not logged 
+    in to perform this action, a `401 Unauthorized` status returned.
+    
+    ```console
+    GET /encounter/:target_encounter_uuid
+    ```
+   
+### Create an encounter
+
+* To Create an encounter you need to specify below attributes in the request body.If you are not logged in to perform this action,
+ a `401 Unauthorized` status returned.
+
+    #### Attributes
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *encounterDatetime* | `Date (ISO8601 Long)` | The date and time the encounter was created (required)
+    *patient* | `Patient UUID` | Patient type resource UUID (required)
+    *encounterType* | `EncounterType UUID` | Encounter type resource UUID (required)
+    *location* | `Location UUID` | Location resource UUID (required)
+    *encounterProviders* | `Array[]: Providers` | Array of Providers already registred in OpenMRS    
```

Comment:
Date type is actually something like: `Array of Provider UUID and associated Encounter Role UUID`

Description: An array of providers and their role within the encounter. At least one provider is required.